### ISSUE TITLE
MongoAcl must has owner for aclService.findChildren function

### ISSUE THREAD

**Issue by knownact** [RovoMe/spring-security-acl-mongodb#1]
unit test:

```java
    @Test
    @WithMockUser
    public void testReadAclsById() throws Exception {
        // Arrange
        TestDomainObject domainObject = new TestDomainObject();
        TestDomainObject firstObject = new TestDomainObject();
        TestDomainObject secondObject = new TestDomainObject();
        TestDomainObject thirdObject = new TestDomainObject();
        TestDomainObject unrelatedObject = new TestDomainObject();

        ObjectIdentity objectIdentity = new ObjectIdentityImpl(Class.forName(domainObject.getClass().getName()), domainObject.getId());

        MongoAcl parent = new MongoAcl(domainObject.getId(), domainObject.getClass().getName(), UUID.randomUUID().toString());
        MongoAcl child1 = new MongoAcl(firstObject.getId(), firstObject.getClass().getName(), UUID.randomUUID().toString(), "Tim Test", parent.getId(), true);
        MongoAcl child2 = new MongoAcl(secondObject.getId(), secondObject.getClass().getName(), UUID.randomUUID().toString(), "Petty Pattern", parent.getId(), true);
        MongoAcl child3 = new MongoAcl(thirdObject.getId(), thirdObject.getClass().getName(), UUID.randomUUID().toString(), "Sam Sample", parent.getId(), true);
        MongoAcl nonChild = new MongoAcl(unrelatedObject.getId(), unrelatedObject.getClass().getName(), UUID.randomUUID().toString());

        DomainObjectPermission permission = new DomainObjectPermission(UUID.randomUUID().toString(),
            SecurityContextHolder.getContext().getAuthentication().getName(),
            BasePermission.READ.getMask() | BasePermission.WRITE.getMask(),
            true, true, true);

        parent.getPermissions().add(permission);
        child1.getPermissions().add(permission);
        child2.getPermissions().add(permission);

        // MongoAcl must has owner

        aclRepository.save(parent);
        aclRepository.save(child1);
        aclRepository.save(child2);
        aclRepository.save(child3);
        aclRepository.save(nonChild);

        // Act
        List<Sid> sids = new LinkedList<>();
        sids.add(new PrincipalSid(SecurityContextHolder.getContext().getAuthentication().getName()));

        List<ObjectIdentity> childObjects = aclService.findChildren(objectIdentity);
        //List<ObjectIdentity> childObjects = aclService.findChildren(new ObjectIdentityImpl(Class.forName(firstObject.getClass().getName()), firstObject.getId()));
        Map<ObjectIdentity, Acl> resultUser = aclService.readAclsById(childObjects, sids);

        // Assert
        assertThat(resultUser.keySet().size(), is(equalTo(1)));
    }
```

and aclService.findChildren has such exception:
```java
java.lang.IllegalArgumentException: Principal required

	at org.springframework.util.Assert.hasText(Assert.java:181)
	at org.springframework.security.acls.domain.PrincipalSid.<init>(PrincipalSid.java:44)
	at org.springframework.security.acls.mongodb.BasicLookupStrategy.convertToAcl(BasicLookupStrategy.java:213)
```

**Comment by RovoMe** [comment]
Thank you for your input. The exception is quite easy to fix by setting the owner of the ACL to the current principal of the security context for the constructor not defining any owner. 

As the code is based on the SQL service with modifications done to the queries to work against a MongoDB instead of any SQL based database, the logic in the SQL service may fail this test as well as the BasicLookupStrategy in the JDBC case (as well as in the MongoDB case) is ignoring passed SIDs in general as can be seen in the Javadocs of [`BasicLookupStrategy::readAclsById`](http://docs.spring.io/spring-security/site/docs/current/apidocs/org/springframework/security/acls/jdbc/BasicLookupStrategy.html#readAclsById-java.util.List-java.util.List-)

> @param sids the SIDs for which identities are required (ignored by this implementation)

As the JDBC service (or its lookup strategy) is ignoring it, I am not sure about the correct semantics. Also, the documentation on Spring Security ACL does not explain the relation on parent-child ACLs well, IMO. As `readAclsById` looksup `ACL`s for provided `ObjectIdentity` instances, the Javadoc of the `AclService` is a bit ambigious in the regards of what to return. In the `@return` part of the Javadoc it clearly states that a map with exactly one element (=ACL) for each ObjectIdentity passed in is returned. The first line of the Javadoc however states that only Acls are returned for the passed SIDs. If a passed in ObjectIdentity does not belong to any principal passed in via the SID, how can the method return an ACL per passed in ObjectIdentity in that case? I will ask for clarification in the Spring Security project.

In the meantime, as you've provided a test-case, which I'll add to the tests, I'm not fully sure what your expectation is on the returned ACLs. Currently it is returning 3 ACLs as 3 ObjectIdentities are passed in, hence your assertion will fail. I am also not sure how only one ACL will ever be returned by this test. If you expected the parent ACL to be returned than the invoked methods are wrong as in the current form only the ACLs for the passed in object identities are collected. Please ping back on the behavior you tried to test.

**Comment by knownact** [comment]
The test case is corrected in #2

**Comment by RovoMe** [comment]
I'm closing this issue as the currently authenticated user is now set as owner in case of non-parent ACLs which should solve the actual issue reported.

In regards to testcase #2 I'm not sure if the current solution (explained in #3) does solve your needs. I'm also not sure if your last comment was related to your original testcode or to my updated one.
